core/data/src/main/java/com/mifos/core/data/pagingSource/ClientChargesPagingSource.kt

```diff
@@ -55,22 +50,16 @@ class ClientChargesPagingSource(
         clientId: Int,
         position: Int,
     ): Pair<List<Charges>, Int> {
-        return suspendCancellableCoroutine { continuation ->
-            dataManagerCharge.getClientCharges(clientId = clientId, offset = position, 10)
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribeOn(Schedulers.io())
-                .subscribe(object : Subscriber<Page<Charges>>() {
-                    override fun onCompleted() {
-                    }
-
-                    override fun onError(exception: Throwable) {
-                        continuation.resumeWithException(exception)
-                    }
+        return try {
+            val page = dataManagerCharge.getClientCharges(
+                clientId = clientId,
+                offset = position,
+                limit = 10,
+            ).first()
 
-                    override fun onNext(page: Page<Charges>) {
-                        continuation.resume(Pair(page.pageItems, page.totalFilteredRecords))
-                    }
-                })
+            Pair(page.pageItems, page.totalFilteredRecords)
+        } catch (exception: Throwable) {
+            throw exception
```

Comment:
I suggest wrapping the throw statement in a custom exception. It will help us to easily trace errors during debugging.
somehting like: 
throw DatabaseFetchException("Failed to fetch client charges", exception)